### PR TITLE
genai: move source of truth of example_test and add code generator

### PR DESCRIPTION
**Pull request by eliben**
We'd like to automatically extract code snippets from our examples for the official documentation; the documentation tooling extracts code snippets delineated with special [START... and [END... markers.

However, we also want our examples to appear on the pkgsite without these special comments.

Therefore, the source of truth for example_test is moved to an internal directory from which snippets will be extracted; a tool processes this file to remove these special comments before placing it in genai/example_test.go

One snippet is added as an example - context window reporting for tokens

### CODE REVIEW

File: genai/internal/cmd/gen-examples/gen-examples.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// limitations under the License.
+
+// This code generator takes examples from the internal/snippets directory
+// and copies them to "official" examples in genai/example_test.go, while
+// removing snippet comments (between [START...] and [END...]) that are used
+// for website documentation purposes.
+//
+// Run from the root directory:
+//
+//   go run ./genai/internal/cmd/gen-examples/ < genai/internal/snippets/example_test.go > genai/example_test.go
```
Comment (jba) [inline genai/internal/cmd/gen-examples/gen-examples.go:22]: Add a go:generate in the root to do this?

**Review by jba — APPROVED** [review]

File: genai/internal/cmd/gen-examples/gen-examples.go (hunk 1 above)
Comment (eliben) [reply genai/internal/cmd/gen-examples/gen-examples.go:22]: What do you mean by "in the root"? The origin file? I was worried that it gets copied and then the destination file will have the same go:generate line which may run again
I suppose I can tweak the tool to remove the go:generate line when copying; what do you think?

Comment (jba) [reply genai/internal/cmd/gen-examples/gen-examples.go:22]: I wasn't clear. I was thinking genai/client.go.
Just so you could call `go generate` in the genai directory and everything would happen.
But maybe that's a bad idea. We don't necessarily want to run protoveneer every time we update the examples.
Could you put it in another .go file in the snippets dir that isn't copied?
I'm also fine just not doing it.

Comment (eliben) [reply genai/internal/cmd/gen-examples/gen-examples.go:22]: I'm going to get this in for now, so we can point them to actual samples in the format they want and get the ball rolling. We can work out the exact mechanics of generation later - I'll think about it more and will send a follow-up

**Pull request merged**